src/main/java/gyro/google/compute/InstanceGroupManagerResource.java

```diff
@@ -0,0 +1,166 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package gyro.google.compute;
+
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import com.google.api.client.util.Data;
+import com.google.api.services.compute.Compute;
+import com.google.api.services.compute.model.InstanceGroupManager;
+import com.google.api.services.compute.model.InstanceGroupManagerAutoHealingPolicy;
+import com.google.api.services.compute.model.InstanceGroupManagersSetInstanceTemplateRequest;
+import com.google.api.services.compute.model.Operation;
+import gyro.core.GyroUI;
+import gyro.core.Type;
+import gyro.core.resource.Immutable;
+import gyro.core.resource.Resource;
+import gyro.core.scope.State;
+import gyro.core.validation.Required;
+
+/**
+ * Creates an instance group manager.
+ *
+ * Example
+ * -------
+ *
+ * .. code-block:: gyro
+ *
+ *     google::compute-instance-group-manager instance-group-manager-example
+ *         name: "instance-group-manager-example"
+ *         base-instance-name: "instance-group-manager-example"
+ *         description: "Instance group manager example"
+ *         instance-template: $(google::compute-instance-template instance-group-template-example)
+ *         target-size: 1
+ *         zone: "us-central-1"
+ *     end
+ */
+@Type("compute-instance-group-manager")
+public class InstanceGroupManagerResource extends AbstractInstanceGroupManagerResource {
+
+    private String zone;
+
+    /**
+     * The zone where the managed instance group is located.
+     */
+    @Immutable
+    @Required
+    public String getZone() {
+        return zone;
+    }
+
+    public void setZone(String zone) {
+        this.zone = zone;
+    }
+
+    @Override
+    public boolean doRefresh() throws Exception {
+        Compute client = createClient(Compute.class);
+        copyFrom(client.instanceGroupManagers().get(getProjectId(), getZone(), getName()).execute());
+        return true;
+    }
+
+    @Override
+    public void doCreate(GyroUI ui, State state) throws Exception {
+        InstanceGroupManager instanceGroupManager = createInstanceGroupManager();
+        Compute client = createClient(Compute.class);
+        Operation operation = client.instanceGroupManagers()
+            .insert(getProjectId(), getZone(), instanceGroupManager)
+            .execute();
+        waitForCompletion(client, operation);
+        refresh();
+    }
+
+    @Override
+    public void doUpdate(GyroUI ui, State state, Resource current, Set<String> changedFieldNames)
+        throws Exception {
+        // TODO: target pool resource
+        // https://github.com/perfectsense/gyro-google-provider/issues/79
+        //    private List<AbstractTargetPoolResource> targetPools;
```

Comment:
Should be removed and be done as a separate PR when target pool is available

Reply:
Yeah, it should be handled in a separate PR, but better to leave the comment as it will be helpful when implementing it.